lib/replicator.js

```diff
@@ -2275,12 +2295,15 @@ module.exports = class Replicator {
     protomux.stream.setMaxListeners(0)
     protomux.stream.on('close', this._onstreamclose)
 
+    if (useSession) this._peerSessions++
     this._ifAvailable++
+
     protomux.stream.opened.then((opened) => {
+      if (useSession) this._peerSessions--
       this._ifAvailable--
 
       if (opened && !this.destroyed) makePeer()
```

Comment:
Sanity check: no need to check `this.core.closed` here?

Reply:
prop should be unrelated to this i think